opm/input/eclipse/Schedule/Group/Group.hpp

```diff
@@ -174,6 +174,31 @@ struct GroupInjectionProperties {
     }
 };
 
+struct GroupLimitAction
+{
+    ExceedAction allRates{ExceedAction::NONE};
```

Comment:
Leaving comments when investigating the PR https://github.com/OPM/opm-simulators/pull/5432  for records. Not meaning the owner of the PR to do anything. 

`allRates` basically mean `water, gas, oil, and liquid`.  Very likely we should use `oil` here instead of `allRates` for easier implementation, otherwise it might be messy since `allRates` and `water, gas and liquid` have overlap.

Reply:
> Very likely we should use `oil` here instead of `allRates` for easier implementation,

No, it's a little more subtle than that.  Item 7 (`allRates`) is the **default** behaviour for all phases, but items 11-13 have **overriding** behaviours for those specific phases&ndash;possibly ignored based on the setting in item 2.  As an example, if item 7 is `WELL` and item 12 is `+CON`, then exceeding either of the oil, water, or liquid limits will plug the worst-offending well, but exceeding the gas limit will shut the worst-offending connections and all those below it.

Reply:
Yeah. You can think this way. It also depend whether the item 3-6 have upper limits set for certain phases (UDQ might complicate this, we do not need to determine this in the parsing stage since it will not be broken anyway).  I was thinking we can get the ultimate action for each phase during the parsing stage. 

For example, if the item 2 is not `GRAT`, and item 7 is `WELL` and item 12 is `+CON`.  I would think actions for `water, oil, liquid` are `WELL` and for `gas` is `+CON`.

Reply:
> To expand a little on the above, we https://github.com/OPM/opm-common/pull/3894#pullrequestreview-1980336468 in a bit more detail in https://github.com/OPM/opm-common/pull/3894 earlier this year.

Thanks for pointing me to here.

Reply:
I mean, `allRates` are actions for four limits, we have individual for three of them already, I kind of prefer not using `allRates` and just use the fourth one directly.  It is still possible that there is advantages to use `allRates` when coming to implementation, but let us check where is some logic loophole/misunderstanding first.

Reply:
I see we are enforcing the logic from the simulator side. 

With the master branch, then we might potentially be enforcing the wrong behavior. 

In the `actionOnBrokenConstraints` function (for production group), if `group_limit_action.allRates == RATE`, then we will not check what is specified in `gas`, `water` or `liquid`, which can potentially overwrite the actions for their individual phases. 

Although, in the current stage, it should not cause problem since we only allow `RATE` or `NONE`.

Reply:
And in the master branch (not necessarily from this PR), we did not update the actions based on the item 2. 

For example, if item 2 is `WRAT`, we did not update the `water` to be `RATE`. 

And also, if item 2 is `ORAT`,  we can not do anything with the current `allRates, water, gas and liquid` setup.